### PR TITLE
Tensorflow Dataloader

### PR DESCRIPTION
**Pull request by soumik12345**
- [x] Add base dataloader
- [x] Add YOLOv1 dataloader
- [x] Add corresponding unit tests
- [ ] Add docstrings

### CODE REVIEW

**Comment by soumik12345** [comment]
@arindas A few points:

- I intentionally left the outputs in the notebook for your convenience, will remove the outputs before merging. Please review the notebook using ReviewNB.
- I will start implementing the label creation after feedback round 1. Note that we can implement the label creation as part of the dataloader as a parse function or as a preprocessing layer as part of the layer. Would really appreciate your thoughts on this.

File: keras_object_detection/dataloader/augmentations.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import tensorflow as tf
+
+
+def random_flip_data(image, boxes, classes):
+    w = tf.cast(tf.shape(image)[1], dtype=tf.float32)
+    if tf.random.uniform(()) > 0.5:
+        image = tf.image.flip_left_right(image)
+        boxes = tf.stack(
```
Comment (arindas) [inline keras_object_detection/dataloader/augmentations.py:10]: Add a comment describing the mirroring of the boxes.

**Review by arindas — CHANGES_REQUESTED** [review]
Check comment in keras_object_detection/dataloader/augmentations.py

**Review by arindas — APPROVED** [review]

File: keras_object_detection/dataloader/yolov1_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        )
+        self.output_dim = self.predictions_per_cell * 5 + self.n_classes
+
+    def preprocess_outputs(self, boxes, labels):
+        boxes = tf.cast(boxes, dtype=tf.float32)
+        boxes_xywh = box_utils.convert_to_xywh(boxes)
+        classes = tf.one_hot(labels, depth=self.n_classes, dtype=tf.float32)
+        num_objects = tf.shape(classes)[0]
+        pc = tf.ones(shape=[num_objects, 1], dtype=tf.float32)
+        box_centers = boxes_xywh[:, :2]
```
Comment (arindas) [inline keras_object_detection/dataloader/yolov1_dataloader.py:60]: x + w/2, y + h/2 where?
